README.md

```diff
@@ -9,10 +9,100 @@ inspect, and authenticate Bitcoin transactions.
 ### IMPORTANT WARNING
 
 It is extremely easy to write insecure code using these libraries. We do not
-espouse a specific security model. Any SPV verification involves complex
+recommend a specific security model. Any SPV verification involves complex
 security assumptions. Please seek external review for your design before
 building with these libraries.
 
+### Solidity Compiler
+
+Starting from version `1.1.0`, required solidity compiler (`solc`) version is
+at least `0.5.10`.
+
+### How are proofs formatted?
+
+An SPV interaction has two players: a prover and a verifier. The prover submits
+an SPV proof, and the verifier checks it.
+
+The proof must contain several elements: a transaction, an inclusion proof, and
+a header chain. For convenience and gas minimization, we have a standard format
+for these:
+
+1. The transaction is pre-parsed by the prover into:
+    1. The transaction version (currently always 1 or 2 as a 4-byte LE integer)
+    1. The variable-length input vector
+        1. Tightly packed in a single bytearray called `vin`
+    1. The variable-length output vector
+        1. Tightly packed in a single bytearray called `vout`
+    1. The transaction locktime (a 4-byte LE integer)
+1. The header chain:
+    1. Contains any number of 80-byte headers
+    1. Is a single bytearray without prefix or padding
+    1. Starts from the lowest height
```

Comment:
Generally yes, but this is an implementation detail. The library is less opinionated about it than it used to be. In some uses this may not be the case (and in relays, there may be no transaction involved)